functions/transformations/difference.go

```diff
@@ -178,68 +180,68 @@ func (t *differenceTransformation) Process(id execute.DatasetID, tbl flux.Table)
 
 	// We need to drop the first row since its derivative is undefined
 	firstIdx := 1
-	return tbl.Do(func(cr flux.ColReader) error {
+	return tbl.DoArrow(func(cr flux.ArrowColReader) error {
 		l := cr.Len()
 
 		if l != 0 {
 			for j, c := range cols {
 				d := differences[j]
 				switch c.Type {
 				case flux.TBool:
-					if err := builder.AppendBools(j, cr.Bools(j)[firstIdx:]); err != nil {
+					if err := appendSlicedCol(j, firstIdx, c.Type, cr, builder); err != nil {
```

Comment:
Instead of `appendSlicedCol` could you not have just used the `arrow.AppendBools` utility directly here?